reorder_python_imports.py

```diff
@@ -244,6 +244,17 @@ def _inner() -> Generator[CodePartition, None, None]:
                         new_src = import_obj.to_text()
                         yield partition._replace(src=new_src)
                         break
+                    # from a.b.c import d => from c import d
+                    elif (
+                        not attr and
+                        (mod_parts + [symbol] == orig_mod) and
+                        len(new_mod) > 1 and
+                        symbol == new_mod[-1]
+                    ):
```

Comment:
there's some fairly complex logic going on here and no tests demonstrating why it's necessary

Reply:
your new test also passes with half of these conditionals commented out

Reply:
Added some more tests, lmk what you think.